tests/generate.py

```diff
@@ -513,7 +513,14 @@ def name_constraints() -> None:
                 )
             ],
         )
-
+    
+        generate_name_constraints_test(
```

Comment:
While we're testing something related to certificate names there's nothing here specific to name **constraints**.

It feels confusing to add an unrelated test case to the name constraint unit tests. I think for this one-off test I would be happy with a single Rust unit test that calls `verify_is_valid_for_subject_name` with the test end-entity certificate you generated to add to `tests/name_constraints`. E.g. we can skip the Python test generator part, check-in the generated EE cert (please put it somewhere other than the name constraints dir please), and write a small unit test in `end_entity.rs` to use the test cert. We don't need the CA, and the `verify_is_valid_for_subject_name` code path doesn't check expiry so we don't need to keep the EE cert fresh over time.

If you or Djc are keen on seeing the test case/certificates generated dynamically and dislike my suggestion I think we should do the extra work to make a new "suite" for this test instead of adding it to the name constraint test suite. 

Does that make sense?

Reply:
The argument from B) is compelling to me.  

What about instead of creating a separate "suite" for this test, we rename the name constraint bits in generate.py/associated files to called tls_server_cert tests. The `check_cert` helper the generated tests use in `name_constraints.rs` doesn't seem overly specific to name constraint testing vs generically asserting validity/invalidity of a set of names for a TLS EE server cert. 

With a rename this new test case fits in without issue. I'm happy to look at that separate from this PR, allowing this one to land as-is.

Reply:
> With a rename this new test case fits in without issue. I'm happy to look at that separate from this PR, allowing this one to land as-is.

https://github.com/rustls/webpki/pull/75